### DISCUSSION TITLE
Hide features and subfeatures

### DISCUSSION THREAD

**Discussion by LankyAlec** [GMOD/jbrowse-components#4084] [Q&A] [answered]
Hi, I'm sure this topic has already been addressed but I haven't been able to find the discussion.
What I need is to hide part of the attributes and all the subfeatures from specific tracks when opening the detail of a track element.
Is there a way to do this by inserting some code into the json of the tracks?

i use react-linear-genome-view v.2.8

<img width="329" alt="Screenshot 2023-11-23 alle 10 38 24" src="https://github.com/GMOD/jbrowse-components/assets/66473605/0af3c49c-b215-400e-8086-2e77b38b5e4e">

**Comment by cmdcolin** [comment]
Hi there, I am not sure this is possible with the default featureDetails formatter right now. I will check and see. there is a bug for the undefined thing here, and we could make it so that first level is collapsed by default if it is obtrusive. would that be preferable, or is hiding the goal?

https://github.com/GMOD/jbrowse-components/issues/3950

**Reply by cmdcolin** [reply]
I made a new branch here https://github.com/GMOD/jbrowse-components/pull/4091 that allows you to set

```
formatDetails: {
   maxDepth: 0
}
```
on the track config. it is not released on NPM so not sure you can test it out quite yet, but let me know what you think about that change

**Comment by LankyAlec** [comment]
sorry if I'm only replying now.
it would be a perfect solution for subfeatures.
While to hide specific attributes of the track that are not of interest to the "visitor" such as itemRGB. do you have any solution?

**Reply by cmdcolin** [reply, marked as answer]
the itemRGB can be hidden by adding itemRGB:undefined to a jexl callback for formatDetails->feature

an example is here: it hides the phase field https://github.com/GMOD/jbrowse-components/blob/b0136f11d2b62baf8bd96d2a6b4737e6bbe3e122/test_data/volvox/config.json#L1085


hope that helps!

**Reply by cmdcolin** [reply]
the maxDepth feature was also released in v2.9.0 now (I changed hideSubfeatures to maxDepth to allow the off chance that someone wants to just hide the subSubFeatures, and can thus specify maxDepth 1, but to hide all subfeatures, you can say)

```
formatDetails: {
   maxDepth: 0
}
```

**Reply by LankyAlec** [reply]
Great!!!
